packages/roosterjs-editor-dom/lib/utils/setListItemStyle.ts

```diff
@@ -0,0 +1,58 @@
+import ContentTraverser from '../contentTraverser/ContentTraverser';
+import getStyles from '../style/getStyles';
+import safeInstanceOf from './safeInstanceOf';
+import setStyles from '../style/setStyles';
+import { default as findClosestElementAncestor } from './findClosestElementAncestor';
+import { InlineElement } from 'roosterjs-editor-types';
+
+/**
+ * Set the Style of a List Item provided, with the styles that the inline child elements have
+ * If the child inline elements have different styles, it will not modify the styles of the list item
+ * @param element the LI Element to set the styles
+ * @param styles The styles that should be applied to the element.
+ */
+export default function setListItemStyle(element: HTMLLIElement, styles: string[]) {
+    const elementsStyles = getInlineChildElementsStyle(element);
+    let stylesToApply: Record<string, string> = getStyles(element);
+
+    styles.forEach(styleName => {
+        const styleValues = elementsStyles.map(style =>
+            style[styleName] !== undefined ? style[styleName] : ''
+        );
+
+        if (
+            styleValues &&
+            (styleValues.length == 1 || new Set(styleValues).size == 1) &&
+            styleValues[0]
+        ) {
+            stylesToApply[styleName] = styleValues[0];
+        }
+    });
+    setStyles(element, stylesToApply);
+}
+
+function getInlineChildElementsStyle(element: HTMLElement) {
+    const result: Record<string, string>[] = [];
+    const contentTraverser = ContentTraverser.createBodyTraverser(element);
+    let currentInlineElement: InlineElement;
+
+    while (contentTraverser.currentInlineElement != currentInlineElement) {
+        currentInlineElement = contentTraverser.currentInlineElement;
+        let currentNode = currentInlineElement.getContainerNode();
+
+        if (currentNode.nodeName != 'BR') {
+            if (currentNode.nodeName == '#text') {
```

Comment:
no need to do this check. findClosestElementAncestor will do the check for you

Reply:
I think that if the inline element is a span, the findClosestElementAncestor would return the List Item, and we want to get the styles of the childs of the inline childrens. I think we only need to execute  findClosestElementAncestor  when the node it s text node

Reply:
findClosestElementAncestor already does this for you. If you check here, it means we actually checked twice

Reply:
Removing this check